feat(curriculum): add extension tests for css-flexbox curriculum modules

luqven opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Related #39608

<!-- Feel free to add any additional description of changes below this line -->

Replaces #40272

Changes:

Adds a test that validates the presence of `follow-btn` on the page and that its display property is not set to `none` to the following:

- css-flexbox/add-flex-superpowers-to-the-tweet-embed.md
- css-flexbox/apply-the-flex-direction-property-to-create-a-column-in-the-tweet-embed.md
- css-flexbox/apply-the-flex-direction-property-to-create-rows-in-the-tweet-embed.md
- css-flexbox/use-the-align-items-property-in-the-tweet-embed.md
- css-flexbox/use-the-justify-content-property-in-the-tweet-embed.md

luqven committed:
feat(curriculum): add extension tests for css-flexbox curriculum modules

nhcarrigan commented on curriculum/challenges/english/01-responsive-web-design/css-flexbox/add-flex-superpowers-to-the-tweet-embed.md:
````diff
@@ -54,6 +54,12 @@ Your `.follow-btn` should have a `display` property set to `flex`.
+Your `.follow-btn` should be rendered on the page. Be sure to turn off any extensions such as ad blockers.
+
+```js
+testString: assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
````
```suggestion
assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
```

nhcarrigan commented on curriculum/challenges/english/01-responsive-web-design/css-flexbox/apply-the-flex-direction-property-to-create-a-column-in-the-tweet-embed.md:
````diff
@@ -22,6 +22,12 @@ The `.profile-name` element should have a `flex-direction` property set to colum
+Your `.follow-btn` should be rendered on the page. Be sure to turn off any extensions such as ad blockers.
+
+```js
+testString: assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
````
```suggestion
assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
```

nhcarrigan commented on curriculum/challenges/english/01-responsive-web-design/css-flexbox/apply-the-flex-direction-property-to-create-rows-in-the-tweet-embed.md:
````diff
@@ -28,6 +28,12 @@ The `footer` should have a `flex-direction` property set to row.
+Your `.follow-btn` should be rendered on the page. Be sure to turn off any extensions such as ad blockers.
+
+```js
+testString: assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
````
```suggestion
assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
```

nhcarrigan commented on curriculum/challenges/english/01-responsive-web-design/css-flexbox/use-the-align-items-property-in-the-tweet-embed.md:
````diff
@@ -22,6 +22,12 @@ The `.follow-btn` element should have the `align-items` property set to a value
+Your `.follow-btn` should be rendered on the page. Be sure to turn off any extensions such as ad blockers.
+
+```js
+testString: assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
````
```suggestion
assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
```

nhcarrigan commented on curriculum/challenges/english/01-responsive-web-design/css-flexbox/use-the-justify-content-property-in-the-tweet-embed.md:
````diff
@@ -26,6 +26,12 @@ assert(
+Your `.follow-btn` should be rendered on the page. Be sure to turn off any extensions such as ad blockers.
+
+```js
+testString: assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
````
```suggestion
assert($('.follow-btn').length > 0 && $('.follow-btn').css('display') !== 'none');
```

naomi-lgbt reviewed (changes requested):
Our challenge template has changed - since the test runner is no longer using `YAML` strings, we do not need the `testString` property any more. 🙂

luqven committed:
apply suggestions from code review

Co-authored-by: Nicholas Carrigan (he/him) <nhcarrigan@gmail.com>

luqven commented:
> Our challenge template has changed - since the test runner is no longer using `YAML` strings, we do not need the `testString` property any more. 🙂

thanks ❤️ , still getting used to the changes 🤦🏼. Committed your suggestions.

RandellDawson commented:
@luqven FYI - You can test that a challenge passes all the tests by performing the following commands locally:
```bash
cd curriculum
npm run test -- -g 'title of challenge'
```
If all the tests pass, you at least know pushing the changes should not fail for that reason.

lasjorg commented:
@nhcarrigan is anything still blocking this?

@luqven I would prefer if this didn't close the issue. I'm not convinced we have found the right solution for the issues with dark mode extensions and the color challenges.

nhcarrigan commented:
Yes, we are still in a code freeze and are not merging any PRs to challenge files at this time. 🙂

lasjorg commented:
@nhcarrigan Thanks for pinning the code freeze issue again.

naomi-lgbt reviewed (approved):
LGTM~! 🥳 

Note: In order to test this, I had to force my ad-blocker to hide the element with a custom filter:
![image](https://user-images.githubusercontent.com/63889819/104655560-bb166680-5672-11eb-8e19-05adbada5f2e.png)


Thank you for your pull request! We look forward to additional contributions! 

Please note that we are currently in a code freeze and are not merging PRs to curriculum files for the time being.

nhcarrigan commented:
> I would prefer if this didn't close the issue. I'm not convinced we have found the right solution for the issues with dark mode extensions and the color challenges.

I agree. I went ahead and edited the PR to remove that link.

naomi-lgbt reviewed (changes requested):
Hey @luqven,

Actually, would you mind moving these tests to the top of each file's test suites? That way, if a camper has the correct code but is failing due to an extension, this test will be the first thing they see in the console.

RandellDawson commented:
@luqven It has been a while since we last heard from you.  To get this PR merged, can you please make the changes last requested by @nhcarrigan?

Thank you.

raisedadead commented:
Hey @nhcarrigan is this additional review something that we could take up ourselves in interest of time. We are going to be doing a head branch rename and would like to merge any PRs we absolutely can

nhcarrigan commented:
Yes we can absolutely handle the changes ourselves.🙂

luqven commented:
Hey all, apols for the delay. This got drowned out in hundred of notifications.

I can have this in after 5 pm est today. But please feel free to take it off my hands if need be.

raisedadead commented:
No worries @luqven we should be able to add the changes, we typically do not rush anyone. But we are in the middle of moving from `master` to `main` and hence the urgency. Nick should be able to get this in later.

Thanks for your work on the PR.

Commit:
fix: reorder tests

Moves the test that asserts the presence of the follow button to the
top of the order.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

nhcarrigan commented:
I've gone ahead and pushed the update.

@raisedadead am I good to merge after CI passes?

raisedadead commented:
> am I good to merge after CI passes?

Yes.

nhcarrigan commented:
Thank you for your contribution to the project! 👍
We are happy to accept these changes and look forward to future contributions. 🎉
